compiler/testData/diagnostics/tests/enum/enumSubjectTypeCheck.fir.kt

```diff
@@ -32,9 +32,9 @@ fun useEn2(x: En2) = x
 fun bar(x: Any) {
     if (x is En && x is En2) {
         when (x) {
-            En.A -> useEn(x)
-            En2.D -> useEn2(x)
+            <!INCOMPATIBLE_TYPES!>En.A<!> -> useEn(x)
```

Comment:
FE1.0 does not report this because the type of `x` is `Any` rather than an intersection of `En` and `En2`. It seems the FE1.0 checker is not for this diagnostic is not taking smart cast into consideration. For example, the following code demonstrate the bug in FE1.0:

```
fun foo(x: Any, s: String) {
    if (x is String) {
        x == 1 // FE1.0 thinks this is OK
    }
    s == 1 // FE1.0 reports EQUALITY_NOT_APPLICABLE
}
```